src/neuroconv/datainterfaces/ecephys/neuralynx/neuralynxdatainterface.py

```diff
@@ -118,3 +83,109 @@ def __init__(self, folder_path: FolderPathType, sampling_frequency: float = None
+
+
+def get_filtering_multi_channel(neo_reader) -> List[str]:
+    """
+    Get the filtering metadata from neo reader containing a multiple channels.
+
+    Parameters
+    ----------
+    neo_reader: NeuralynxRawIO
+        Neo IO to extract the filtering metadata from
+
+    Returns
+    -------
+    list:
+        list of (str) json dump of filter parameters of each channel.
+         Uses the mu character, which may cause problems
+        for downstream things that expect ASCII.
+    """
+    neo_annotations = neo_reader.raw_annotations
+
+    # extracting infos of each signal (channel)
+    filter_info = []
+    stream_infos = neo_annotations["blocks"][0]["segments"][0]["signals"]
+    # iterate across streams
+    for stream_info in stream_infos:
+        stream_annotations = stream_info["__array_annotations__"]
+        filter_dict = {k: v for k, v in stream_annotations.items() if k.lower().startswith("dsp")}
+        # iterate across channels in stream
+        for chan_idx in range(len(stream_info["__array_annotations__"]["channel_names"])):
+            channel_filter_dict = {k: v[chan_idx] for k, v in filter_dict.items()}
+            # conversion to string values
+            for key, value in channel_filter_dict.items():
+                if not isinstance(value, str):
+                    value = " ".join(value)
+                channel_filter_dict[key] = value
+            filter_info.append(json.dumps(channel_filter_dict, ensure_ascii=True))
+
+    return filter_info
+
+
+def extract_metadata_single_reader(neo_reader) -> dict:
+    """
+    Extract the session metadata from a NeuralynxRawIO object
+
+    Parameters
+    ----------
+    neo_reader: NeuralynxRawIO object
+        Neo IO to extract the metadata from
+
+    Returns
+    -------
+    dict:
+        dictionary containing the session metadata across channels
+        Uses the mu character, which may cause problems
+        for downstream things that expect ASCII.
+    """
+
+    # check if neuralynx file header objects are present and use these metadata extraction
+    if hasattr(neo_reader, "file_headers"):
+        # use only ncs files as only continuous signals are extracted
+        headers = [header for filename, header in neo_reader.file_headers.items() if filename.lower().endswith(".ncs")]
+
+    # use metadata provided as array_annotations for each channel (neo version <=0.11.0)
+    else:
+        headers = []
+        neo_annotations = neo_reader.raw_annotations
+        for stream_annotations in neo_annotations["blocks"][0]["segments"][0]["signals"]:
+            for chan_idx in range(len(stream_annotations["__array_annotations__"]["channel_names"])):
+                headers.append({k: v[chan_idx] for k, v in stream_annotations["__array_annotations__"].items()})
+
+    # extract common attributes across channels
+    common_header = _dict_intersection(headers)
```

Comment:
Can you mention what this function is doing?

Reply:
I extended the comment. Also the function itself has a docstring.

Reply:
I was wondering more about _dict_intersection, what are the cases when:

```
    if len(dictionaries) == 0:
        return {}
    if len(dictionaries) == 1:
        return dictionaries[0]
  
```

Reply:
All right, I was finding this function confusing, Here is my suggestion of how to make it more readable that also includes type hints:

```
def _dict_intersection(dict_list: List) -> Dict:
    """
    Intersect dict_list and return only common keys and values
    Parameters
    ----------
    dict_list: list of dicitionaries each representing a header
    Returns
    -------
    dict:
        Dictionary containing key-value pairs common to all input dicitionary_list
    """
    if len(dict_list) == 0:
        return {}
    if len(dict_list) == 1:
        return dict_list[0]

    # Collect keys appearing in all dictionaries
    common_keys = set.intersection(*[set(dict.keys()) for dict in dict_list])

    # Add values for common keys if the value is identical across all headers (dict_list)
    first_dict = dict_list[0]
    all_dicts_have_same_value_for = lambda key: all([first_dict[key] == dict[key] for dict in dict_list])
    common_header = {key: first_dict[key] for key in common_keys if all_dicts_have_same_value_for(key)}
    return common_header
```

I am still not sure the cases are necessary. I don't think the first can happen as it will fail before and the second should work anyway with the functions as it is.

Reply:
I don't understand, this is an aux function that should only be used inside this interface (it is marked as private). If we are using it in this interface then it means that the recordings should have continous data or the interface should have failed before. Asked in another way, what would lead to a situation where there would be no headers in the context of this interface? Can you flesh out the case a little bit more so I can think on what's the best way of building a safety test for this as well.

Reply:
The case I have in mind is that at the end you want to run the conversion on many recording sessions, most of which are complete in the sense that among other data they contain continuous neuralynx data. However, knowing the difficulty of having 100% complete datasets there might be a few sessions where the continuous neuralynx data might not be present. In that case,  do you prefer the conversion process to break and raise an error here or do you want to convert as much of the data as you can extract and just leave the continuous data empty? I agree this is kind of a corner case, but for later usage of the package potentially important.

Reply:
OK, I got this. I will discuss this in the next meeting on how to have a common way of dealing with this problem.